src/node/SetupServerApi.ts

```diff
@@ -1,44 +1,59 @@
-import { invariant } from 'outvariant'
 import {
   BatchInterceptor,
   HttpRequestEventMap,
   Interceptor,
   InterceptorReadyState,
 } from '@mswjs/interceptors'
+import { invariant } from 'outvariant'
 import { SetupApi } from '~/core/SetupApi'
 import { RequestHandler } from '~/core/handlers/RequestHandler'
 import { LifeCycleEventsMap, SharedOptions } from '~/core/sharedOptions'
 import { RequiredDeep } from '~/core/typeUtils'
-import { mergeRight } from '~/core/utils/internal/mergeRight'
 import { handleRequest } from '~/core/utils/handleRequest'
 import { devUtils } from '~/core/utils/internal/devUtils'
-import { SetupServer, SetupServerInternalContext } from './glossary'
+import { mergeRight } from '~/core/utils/internal/mergeRight'
+import { SetupServer, SetupServerContext } from './glossary'
 import { isNodeExceptionLike } from './utils/isNodeExceptionLike'
 
 const DEFAULT_LISTEN_OPTIONS: RequiredDeep<SharedOptions> = {
   onUnhandledRequest: 'warn',
 }
 
+type Interceptors = Array<{
+  new (): Interceptor<HttpRequestEventMap>
+}>
+
+const defaultContext = { nodeEvents: undefined }
 export class SetupServerApi
   extends SetupApi<LifeCycleEventsMap>
   implements SetupServer
 {
-  private context: SetupServerInternalContext
+  private declare context: SetupServerContext
   protected readonly interceptor: BatchInterceptor<
     Array<Interceptor<HttpRequestEventMap>>,
     HttpRequestEventMap
   >
   private resolvedOptions: RequiredDeep<SharedOptions>
 
   constructor(
-    interceptors: Array<{
-      new (): Interceptor<HttpRequestEventMap>
-    }>,
+    interceptorsOrConfig:
+      | Interceptors
```

Comment:
I'm not a fan of overly dynamic call signatures. Node has had enough of me. I think your approach to solve this problem in #1879 is better.